### PR TITLE
Feat/net097 20/guards

### PR DESCRIPTION
**Pull request by 5haki6**

### CODE REVIEW

**Review by Alex-knu — APPROVED** [review]

File: MilitaryManager.IdentityServer/Controllers/CommanderController.cs [hunk 1]
```diff
@@ -15,10 +17,13 @@ public class CommanderController : Controller
...
         private readonly CommanderService _commanderService;
-        public CommanderController(UserManager<ApplicationUser> userManager)
+
+        private readonly IHttpContextAccessor _httpContextAccessor;
+        public CommanderController(UserManager<ApplicationUser> userManager, IHttpContextAccessor httpContextAccessor)
         {
             _userManager = userManager;
-            _commanderService = new CommanderService(_userManager);
+            _httpContextAccessor = httpContextAccessor;
+            _commanderService = new CommanderService(_userManager, _httpContextAccessor);
```
Comment (vladtymo) [inline MilitaryManager.IdentityServer/Controllers/CommanderController.cs]: Can you configure the Commander Service in Startup.cs and not create an instance explicitly?

File: MilitaryManager.IdentityServer/Program.cs [hunk 2]
```diff
@@ -40,7 +40,7 @@ public static void Main(string[] args)
                     var result = userManager.CreateAsync(user, PASSWORD).Result;
                     if (result.Succeeded)
                     {
-                        userManager.AddToRoleAsync(user, "Admin");
+                       var creatingResult =  userManager.AddToRoleAsync(user, "Admin").Result;
```
Comment (vladtymo) [inline MilitaryManager.IdentityServer/Program.cs]: You can invoke Wait() method and not to create useless variable.

File: MilitaryManager/client-app/src/app/app.module.ts [hunk 3]
```diff
@@ -28,11 +28,11 @@ import { AddCommanderComponent } from './components/add-commander/add-commander.
     BrowserModule,
     AppRoutingModule,
     PrimeNgComponentsModule,
-    AppLayoutModule
+    AppLayoutModule,
```
Comment (vladtymo) [inline MilitaryManager/client-app/src/app/app.module.ts]: What is the reason?

File: MilitaryManager/client-app/src/app/guards/AdminGuard.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+import {Injectable} from "@angular/core";
+
+@Injectable()
+export class AdminGuard implements CanActivate {
+  constructor(private _identityService : IdentityService) {
+  }
+  canActivate() {
+   return this.getRole();
+  }
+  getRole() : Promise<boolean> {
```
Comment (vladtymo) [inline MilitaryManager/client-app/src/app/guards/AdminGuard.ts]: Better to rename this method to 'IsAdmin' or something like that.

File: MilitaryManager/client-app/src/app/guards/AdminGuard.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    return new Promise((resolve, reject) =>
+    {
+      this._identityService.single.get().subscribe( (data:any) => {
+        if (data["0"] == "Admin") {
+          resolve(true);
+        } else {
+          alert("Немає прав");
```
Comment (vladtymo) [inline MilitaryManager/client-app/src/app/guards/AdminGuard.ts:19]: Why don't you return false?

**Review by vladtymo — CHANGES_REQUESTED** [review]

**Review by vladtymo — APPROVED** [review]

**Pull request merged**